### DISCUSSION TITLE
Not found (404) page layout not working at production

### DISCUSSION THREAD

**Discussion by SoA432** [vercel/next.js#55970] [App Router] [answered]
`Version`: `"next": "13.5.3"`
`Structure`: `App directory` with `[...not_found] - page.tsx` for dymamic routes and `not-found.tsx` at root folder.

Everything works as expected `locally` but not working at `production`, instead of `custom 404` got default `500 error` page

Vercel log once i navigate to non-existing route:

```Error: Cannot find module '/var/task/.next/server/app/page.js'
Require stack:
- /var/task/node_modules/next/dist/server/require.js
- /var/task/node_modules/next/dist/server/next-server.js
- /var/task/___next_launcher.cjs
    at Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)
    at /var/task/node_modules/next/dist/server/require-hook.js:54:36
    at Module._load (node:internal/modules/cjs/loader:922:27)
    at e.<computed>._module.Module._load (/var/task/___vc/__launcher.js:15:1964)
    at Module.require (node:internal/modules/cjs/loader:1143:19)
    at mod.require (/var/task/node_modules/next/dist/server/require-hook.js:64:28)
    at require (node:internal/modules/cjs/helpers:121:18)
    at requirePage (/var/task/node_modules/next/dist/server/require.js:109:84)
    at /var/task/node_modules/next/dist/server/load-components.js:59:84
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/var/task/node_modules/next/dist/server/require.js',
    '/var/task/node_modules/next/dist/server/next-server.js',
    '/var/task/___next_launcher.cjs'
  ],
  page: '/'
}
Error: Cannot find module '/var/task/.next/server/app/page.js'
Require stack:
- /var/task/node_modules/next/dist/server/require.js
- /var/task/node_modules/next/dist/server/next-server.js
- /var/task/___next_launcher.cjs
    at Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)
    at /var/task/node_modules/next/dist/server/require-hook.js:54:36
    at Module._load (node:internal/modules/cjs/loader:922:27)
    at e.<computed>._module.Module._load (/var/task/___vc/__launcher.js:15:1964)
    at Module.require (node:internal/modules/cjs/loader:1143:19)
    at mod.require (/var/task/node_modules/next/dist/server/require-hook.js:64:28)
    at require (node:internal/modules/cjs/helpers:121:18)
    at requirePage (/var/task/node_modules/next/dist/server/require.js:109:84)
    at /var/task/node_modules/next/dist/server/load-components.js:59:84
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/var/task/node_modules/next/dist/server/require.js',
    '/var/task/node_modules/next/dist/server/next-server.js',
    '/var/task/___next_launcher.cjs'
  ],
  page: '/'
}
RequestId: 11bac1b7-adb5-47f7-a0e5-fc831a52991a Error: Runtime exited with error: exit status 1
Runtime.ExitError
```
<img width="212" alt="Снимок экрана 2023-09-25 в 20 21 24" src="https://github.com/vercel/next.js/assets/33688931/84ed56c3-d38d-45af-a4c5-89651e856c5d">
<img width="398" alt="Снимок экрана 2023-09-25 в 19 15 05" src="https://github.com/vercel/next.js/assets/33688931/ecb5009e-1d2b-4f80-a2c8-55027270c716">
<img width="841" alt="Снимок экрана 2023-09-25 в 19 15 45" src="https://github.com/vercel/next.js/assets/33688931/8f1da42e-880a-4edf-8b91-b6562a4d33b3">
<img width="489" alt="Снимок экрана 2023-09-25 в 20 18 13" src="https://github.com/vercel/next.js/assets/33688931/ab30f99b-fc1a-41cd-9368-87da14278e30">

**Comment by SoA432** [comment, marked as answer]
Folder structure caused issues to me:
-app
--(some_folder_wrapper) > [dynamicID] > rest pages
--[...not_found]

so (some_folder_wrapper) its just wrapper so there was 2 dynamic routes handlers on same level [dynamicID] and [...not_found], hard to find what is wrong without any errors and everything works locally

**Reply by johnhm22** [reply]
Do you need a dynamic [..not_found] folder as well as the app/not-found.tsx file?
Next.js v13 provides the feature for a customised 404 by means of the not-found.tsx file. It makes no mention of needing a dynamic route was well.
I ask because in dev on my local machine the not-found.tsx works perfectly. When  deployed to Vercel it doesn't and the standard 404 page is displayed.

**Reply by SoA432** [reply]
@johnhm22 here reference https://stackoverflow.com/questions/75302340/not-found-page-does-not-work-in-next-js-13

**Reply by johnhm22** [reply]
Thanks.
I have added in a dynamic [...not-found] folder and page.tsx, deployed on Vercel and it now successfully redirects to my customised 404 page.
